src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.ts

```diff
@@ -40,10 +62,48 @@ export class ParticipationSubmissionComponent implements OnInit {
      * Set up page by loading participation and all submissions
      */
     setupPage() {
+        this.isLoading = true;
         this.route.params.subscribe((params) => {
             this.participationId = +params['participationId'];
+            this.exerciseService.find(params['exerciseId']).subscribe((exerciseResponse) => {
+                this.exercise = exerciseResponse.body!;
+                this.exerciseStatusBadge = moment(this.exercise.dueDate!).isBefore(moment()) ? 'badge-danger' : 'badge-success';
+                if (this.exercise.type === ExerciseType.PROGRAMMING) {
+                    // Get template and solution participation and submission for prog. exercise
+                    this.programmingExerciseService.findWithTemplateAndSolutionParticipation(this.exercise.id!, true).subscribe((response) => {
```

Comment:
I don't like this implementation, it means Artemis will always fetch template and solution participation.
Can we not distinguish earlier which kind of data is necessary and only fetch template or solution or student participation (each with submissions and results)?

I think we should use some element in the URL of the page to indicate what kind of participation it is